coreweave/cks/data_cluster.go

```diff
@@ -0,0 +1,273 @@
+package cks
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	cksv1beta1 "buf.build/gen/go/coreweave/cks/protocolbuffers/go/coreweave/cks/v1beta1"
+	"connectrpc.com/connect"
+	"github.com/coreweave/terraform-provider-coreweave/coreweave"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var (
+	_ datasource.DataSource = &ClusterDataSource{}
+)
+
+func NewClusterDataSource() datasource.DataSource {
+	return &ClusterDataSource{}
+}
+
+type ClusterDataSource struct {
+	client *coreweave.Client
+}
+
+type ClusterDataSourceModel struct {
```

Comment:
Is there a reason we can't use the already existing ClusterResourceModel? That way we can cut down a lot of this duplication

Reply:
I recalled reading something that suggested that decoupling models between resources and data sources was the preferred method... but can't find it now. Combining the models would have the benefit of failing tests for the data source if the model is updated, but the schema is not, so I think you're right, that would be better. It's also an easily reversible decision. Will implement.